cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -1655,16 +1759,83 @@ class self_comparator {
    * @return A binary callable object
    */
   template <bool has_nested_columns,
+            template <typename> typename dispatch_conditional_t = type_identity_t,
             typename Nullate,
             typename PhysicalEqualityComparator = nan_equal_physical_equality_comparator>
   auto equal_to(Nullate nullate                       = {},
                 null_equality nulls_are_equal         = null_equality::EQUAL,
                 PhysicalEqualityComparator comparator = {}) const noexcept
   {
-    return device_row_comparator<has_nested_columns, Nullate, PhysicalEqualityComparator>{
+    return device_row_comparator<has_nested_columns,
+                                 Nullate,
+                                 PhysicalEqualityComparator,
+                                 dispatch_conditional_t>{
       nullate, *d_t, *d_t, nulls_are_equal, comparator};
   }
 
+  /**
+   * @brief Get the variant of comparison operator to use on the device
+   *
+   * Returns a binary callable variant, `F`, with signature `bool F(size_type, size_type)`.
+   *
+   * `F(i,j)` returns true if and only if row `i` compares equal to row `j`.
+   *
+   * @tparam Nullate A cudf::nullate type describing whether to check for nulls.
+   * @tparam PhysicalEqualityComparator A equality comparator functor that compares individual
+   * values rather than logical elements, defaults to a comparator for which `NaN == NaN`.
+   * @param column_types Column types in the row to be compared
+   * @param nullate Indicates if any input column contains nulls.
+   * @param nulls_are_equal Indicates if nulls are equal.
+   * @param comparator Physical element equality comparison functor.
+   * @return A binary callable object variant
+   */
+  template <typename Nullate,
+            typename PhysicalEqualityComparator = nan_equal_physical_equality_comparator>
+  auto equal_to(std::unordered_set<cudf::type_id> column_types,
+                Nullate nullate                       = {},
+                null_equality nulls_are_equal         = null_equality::EQUAL,
+                PhysicalEqualityComparator comparator = {}) const noexcept
+  {
+    using row_comparator_t = std::variant<
+      device_row_comparator<true, Nullate, PhysicalEqualityComparator, type_identity_t>,
+      device_row_comparator<false, Nullate, PhysicalEqualityComparator, dispatch_void_if_nested_t>,
+      device_row_comparator<false,
+                            Nullate,
+                            PhysicalEqualityComparator,
+                            dispatch_void_if_compound_t>>;
+
+    auto find_any = [](std::initializer_list<cudf::type_id> ids,
```

Comment:
```suggestion
    auto any_of = [](std::initializer_list<cudf::type_id> ids,
```
not a hard requirement

Reply:
I can see your concern. The reason that I'm suggesting `any_of` is because we are basically doing a `std::ranges::any_of` and the range is the hash set. Once we migrate to C++ 20. this will be a single call like, e.g. :
```cpp
std::ranges::any_of(column_types, some_pred);
```